se_denseslam/include/se/config.h

```diff
@@ -38,37 +38,162 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 struct Configuration {
 
-	// 
+  //
   // KFusion configuration parameters
-  // Command line arguments are parsed in default_parameters.h 
+  // Command line arguments are parsed in default_parameters.h
   //
 
-	int compute_size_ratio;
-	int integration_rate;
-	int rendering_rate;
-	int tracking_rate;
-	uint3 volume_resolution;
-	float3 volume_size;
-    int voxel_block_size;
-	float3 initial_pos_factor;
-	std::vector<int> pyramid;
-	std::string dump_volume_file;
-	std::string input_file;
-	std::string log_file;
+  /**
+   * The ratio of the input frame size over the frame size used internally.
+   * Values greater than 1 result in the input frames being downsampled
+   * before processing. Valid values are 1, 2, 4 and 8.
+   * <br>\em Default: 1
+   */
+  int compute_size_ratio;
+
+  /**
+   * Perform tracking on a frame every tracking_rate frames.
+   * <br>\em Default: 1
+   */
+  int tracking_rate;
+
+  /**
+   * Integrate a 3D reconstruction every integration_rate frames. Should not
+   * be less than tracking_rate.
+   * <br>\em Default: 2
+   */
+  int integration_rate;
+
+  /**
+   * Render the 3D reconstruction every rendering_rate frames.
+   * <br>\em Default: 4
+   */
+  int rendering_rate;
+
+  /**
+   * The x, y and z resolution of the reconstructed volume in voxels.
+   * <br>\em Default: (256, 256, 256)
+   */
+  uint3 volume_resolution;
+
+  /**
+   * The x, y and z dimensions of the reconstructed volume in meters.
+   * <br>\em Default: (2, 2, 2)
+   */
+  float3 volume_size;
+
+  int voxel_block_size;
+
+  /*
+   * TODO
+   * <br>\em Default: (0.5, 0.5, 0)
+   */
+  float3 initial_pos_factor;
+
+  /**
+   * TODO
+   * <br>\em Default: (10, 5, 4)
+   */
+  std::vector<int> pyramid;
+
+  /*
+   * TODO
+   * <br>\em Default: ""
+   */
+  std::string dump_volume_file;
+
+  /*
+   * TODO
+   * <br>\em Default: ""
+   */
+  std::string input_file;
+
+  /*
+   * TODO
+   * <br>\em Default: ""
+   */
+  std::string log_file;
+
+  /*
+   * TODO
+   * <br>\em Default: ""
+   */
   std::string groundtruth_file;
 
-	float4 camera;
-	bool camera_overrided;
+  /**
+   * The intrinsic camera parameters. camera.x, camera.y, camera.z and
+   * camera.w are the x-axis focal length, y-axis focal length, horizontal
+   * resolution (pixels) and vertical resolution (pixels) respectively.
+   */
+  float4 camera;
+
+  /**
+   * Whether the default intrinsic camera parameters have been overriden.
+   */
+  bool camera_overrided;
+
+  /**
+   * The TSDF truncation bound. Values of the TSDF are assumed to be in the
+   * interval ±mu. See Section 3.3 of \cite NewcombeISMAR2011 for more
+   * details.
+   *  <br>\em Default: 0.1
+   */
+  float mu;
+
+  /*
+   * TODO
+   *
+   * @note Must be non-negative.
```

Comment:
These are mainly SLAMBench 1.0 features, we might actually think of removing them as they clutter the options and in practice are rarely used.